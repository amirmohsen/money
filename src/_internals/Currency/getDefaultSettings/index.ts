import { CurrencySettings } from '..';

const getDefaultSettings = (code: string): CurrencySettings => ({
  code,
  symbol: code,
  thousandsSeparator: ',',
  decimalSeparator: '.',
  decimalDigits: 2,
  pattern: '%ns%s%v',
});

export default getDefaultSettings;
