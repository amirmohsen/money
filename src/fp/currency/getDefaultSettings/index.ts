import { BaseCurrencySettings } from 'src/fp/types';

const getDefaultSettings = (code: string): BaseCurrencySettings => ({
  symbol: code,
  thousandsSeparator: ',',
  decimalSeparator: '.',
  decimalDigits: 2,
  pattern: '%ns%s%v',
});

export default getDefaultSettings;
