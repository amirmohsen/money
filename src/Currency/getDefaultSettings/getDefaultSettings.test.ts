import getDefaultSettings from '.';

test('getDefaultSettings() should match the snapshot', () => {
  expect(getDefaultSettings('USD')).toMatchSnapshot();
  expect(getDefaultSettings('GBP')).toMatchSnapshot();
});
