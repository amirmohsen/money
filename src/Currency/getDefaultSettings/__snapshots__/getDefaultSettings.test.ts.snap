// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDefaultSettings() should match the snapshot 1`] = `
Object {
  "code": "USD",
  "decimalDigits": 2,
  "decimalSeparator": ".",
  "pattern": "%ns%s%v",
  "symbol": "USD",
  "thousandsSeparator": ",",
}
`;

exports[`getDefaultSettings() should match the snapshot 2`] = `
Object {
  "code": "GBP",
  "decimalDigits": 2,
  "decimalSeparator": ".",
  "pattern": "%ns%s%v",
  "symbol": "GBP",
  "thousandsSeparator": ",",
}
`;
